substratevm/src/com.oracle.svm.agent/src/com/oracle/svm/agent/BreakpointInterceptor.java

```diff
@@ -1119,35 +1140,24 @@ private static boolean objectStreamClassConstructor(JNIEnvironment jni, Breakpoi
      */
     private static boolean customTargetConstructorSerialization(JNIEnvironment jni, @SuppressWarnings("unused") Breakpoint bp, InterceptedState state) {
         JNIObjectHandle serializeTargetClass = getObjectArgument(1);
-        String serializeTargetClassName = getClassNameOrNull(jni, serializeTargetClass);
-
-        // Skip Lambda class serialization.
-        if (serializeTargetClassName.contains("$$Lambda$")) {
-            return true;
-        }
+        if (isSerializable(jni, serializeTargetClass)) {
```

Comment:
When serializing lambda class, it's capturing class is also being registered for the serialization since the method com.oracle.svm.agent.BreakpointInterceptor#objectStreamClassConstructor is being called for that class too. In some cases, that class might not be serializable. We simply don't want to add it to the configuration file. Everything would work, but we want to avoid warning in the com.oracle.svm.reflect.serialize.hosted.SerializationFeature$SerializationBuilder#registerWithTargetConstructorClass.